python/cudf/cudf/core/multiindex.py

```diff
@@ -146,10 +134,9 @@ def __init__(
             self._codes = codes
         elif len(levels) == len(codes):
             self._codes = cudf.DataFrame()
-            for i, codes in enumerate(codes):
-                name = column_names[i] or i
-                codes = column.as_column(codes)
-                self._codes[name] = codes.astype(np.int64)
+            for i, code in enumerate(codes):
+                code = column.as_column(code)
+                self._codes[i] = code.astype(np.int64)
```

Comment:
Does this work instead (also deleting the line above)?
```suggestion
            self._codes = cudf.DataFrame._from_data({
                i: column.as_column(code).astype(np.int64)
                for i, code in enumerate(codes)
            })
```